### PR TITLE
refactor(CL): replace 6 return values in CreatePosition with a struct

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

`CreatePosition` has too many return values. This PR refactors 6 of them with a struct.

A separate struct is created (instead of using `model.Position` or similar) so that all fields of the returned struct are guaranteed to be set without redundancy with respect to the input values.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

**Comment by p0mvn** [comment]
devbot add changelog refactor(CL): replace 6 return values in CreatePosition with a struct

**Comment by p0mvn** [comment]
dev-bot add changelog refactor(CL): replace 6 return values in CreatePosition with a struct

**Comment by p0mvn** [comment]
dev-bot add changelog api replace 6 return values in CL CreatePosition with a struct

**Comment by p0mvn** [comment]
Test failure here is coming from `main` and unrelated to this PR

**Comment by ValarDragon** [comment]
devbot add changelog misc allow testing with different chain-id's in E2E testing

**Comment by ValarDragon** [comment]
devbot add changelog api replace 6 return values in CL CreatePosition with a struct

File: CHANGELOG.md [hunk 1]
```diff
@@ -79,6 +83,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#5890](https://github.com/osmosis-labs/osmosis/pull/5890) feat: CreateCLPool & LinkCFMMtoCL pool into one gov-prop
 * [#5964](https://github.com/osmosis-labs/osmosis/pull/5964) fix e2e test concurrency bugs
 * [#5948] (https://github.com/osmosis-labs/osmosis/pull/5948) Parameterizing Pool Type Information in Protorev
+* [#5983](https://github.com/osmosis-labs/osmosis/pull/5983) allow testing with different chain-id's in E2E testing
+
+### API Breaking
+
+* [#5983] (https://github.com/osmosis-labs/osmosis/pull/5983) refactor(CL): replace 6 return values in CreatePosition with a struct
```
Comment (ValarDragon) [inline CHANGELOG.md:90]:
```suggestion
```

File: x/concentrated-liquidity/lp.go [hunk 2]
```diff
@@ -15,6 +15,16 @@ import (
 
 const noUnderlyingLockId = uint64(0)
 
+// CreatePositionData reprsents the return data from CreatePosition.
```
Comment (ValarDragon) [inline x/concentrated-liquidity/lp.go:18]:
```suggestion
// CreatePositionData represents the return data from CreatePosition.
```

interesting the spell check linter didn't catch that

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**